### PR TITLE
DuckPlayer page - mobile support

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/1201141132935289/1208258427491316/f

## DuckPlayer Page (mobile)

- npm run build
- npm run serve-special-pages
- The, navigate to 
   - iOS: http://localhost:3221/duckplayer/index.html?platform=ios
   - macOS: http://localhost:3221/duckplayer/index.html?platform=macOS

Note: 
- Both have been extensively tested on all affected platforms during the ship review

### CODE REVIEW

**Comment by shakyShane** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1028?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#993** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/993?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#1008](https://github.com/duckduckgo/content-scope-scripts/pull/1008) <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1008?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1022](https://github.com/duckduckgo/content-scope-scripts/pull/1022) <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1022?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1028** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1028?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1027** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1027?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "posttest-int": "npm run test-int --workspaces --if-present",
     "posttest-int-x": "npm run test-int-x --workspaces --if-present",
     "serve": "http-server -c-1 --port 3220 integration-test/test-pages",
+    "serve-special-pages": "http-server -c-1 --port 3221 build/integration/pages",
```
Comment (mgurgel) [inline package.json:42]: Very nice

File: packages/special-pages/pages/duckplayer/app/index.js [hunk 2]
```diff
@@ -87,18 +77,23 @@ export async function init (messaging, baseEnvironment) {
...
-                                    <WillThrow />
-                                </UserValuesProvider>
-                            </SettingsProvider>
-                        </MessagingContext.Provider>
-                    </TranslationProvider>
+                    <MessagingContext.Provider value={messaging}>
+                        <SettingsProvider settings={settings}>
+                            <UserValuesProvider initial={init.userValues}>
+                                {settings.layout === 'desktop' && (
+                                    <TranslationProvider translationObject={enStrings} fallback={enStrings} textLength={environment.textLength}>
```
Comment (shakyShane) [inline packages/special-pages/pages/duckplayer/app/index.js:84]: This is deliberately NOT using the translations on desktop yet,

File: packages/special-pages/pages/duckplayer/src/js/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import 'preact/devtools'
```
Comment (mgurgel) [inline packages/special-pages/pages/duckplayer/src/js/index.js]: Can this be added to dev builds only?

File: packages/special-pages/pages/duckplayer/app/providers/OrientationProvider.jsx [hunk 4]
```diff
@@ -1,36 +1,55 @@
...
+        screen.orientation.addEventListener('change', handleOrientationChange);
+        return () => screen.orientation.removeEventListener('change', handleOrientationChange);
+    }, []);
 
     useEffect(() => {
-        const listener = (e) => setTheme(window.innerWidth > window.innerHeight ? 'landscape' : 'portrait')
+        let timer;
+        const listener = () => {
+            clearTimeout(timer);
+            timer = setTimeout(() => onChange(getOrientationFromWidth()), 300)
```
Comment (mgurgel) [inline packages/special-pages/pages/duckplayer/app/providers/OrientationProvider.jsx:23]: Why the 300s timeout on screen resize? Is it to allow for the orientation change to conclude?

File: packages/special-pages/pages/duckplayer/src/js/index.js (hunk 3 above)
Comment (shakyShane) [reply packages/special-pages/pages/duckplayer/src/js/index.js]: oops! yeah let's remove it for now, and then add a esbuild thing to ensure we don't include it by mistake

File: packages/special-pages/pages/duckplayer/app/providers/OrientationProvider.jsx (hunk 4 above)
Comment (shakyShane) [reply packages/special-pages/pages/duckplayer/app/providers/OrientationProvider.jsx:23]: It's just a debounce because 'resize' events are noisy

**Review by mgurgel — APPROVED** [review]
Excellent! 🚢

**Pull request merged**